### PR TITLE
rollout bucketing changed

### PR DESCRIPTION
**Pull request by AbdurRafay**

### CODE REVIEW

**Review by wangjoshuah — DISMISSED** [review]
add a test using a final rollout rule with an audience to make sure the user needs to pass the audience to get a variation in the fall back rule

**Comment by mikeproeng37** [comment]
build

File: OptimizelySDKCore/OptimizelySDKCoreTests/OPTLYDecisionServiceTest.m [hunk 1]
```diff
@@ -643,57 +643,84 @@ - (void)testGetVariationForFeatureWithTargetingRuleBucketing {
...
-    XCTAssertNotNil(decision, @"Get variation for feature with rollout having everyone else rule after failing all targeting rules should return variation: %@", decision);
+    XCTAssertNotNil(decision, @"Get variation for feature with rollout having fall back rule after failing all targeting rules should return variation: %@", decision);
     XCTAssertEqualObjects(decision.variation, expectedDecision.variation);
     
-    OCMVerify([bucketerMock bucketExperiment:everyoneElseRule withBucketingId:kUserId]);
+    OCMVerify([bucketerMock bucketExperiment:fallBackRule withBucketingId:kUserId]);
     [bucketerMock stopMocking];
 }
 
+- (void)testGetVariationForFeatureWithFallBackRuleBucketingId {
```
Comment (mikeng13) [inline OptimizelySDKCore/OptimizelySDKCoreTests/OPTLYDecisionServiceTest.m]: I don't think this test is necessary as we shouldn't be respecting bucketingIds in rollouts

**Review by mikeproeng37 — APPROVED** [review]
lgtm

**Comment by kellyroach-optimizely** [comment]
This is for OASIS-2438

**Comment by kellyroach-optimizely** [comment]
TEST
* sh ./Scripts/build_all.sh PASSED
* 4x Product > Analyze on 4 framework targets PASSED
* 8+1 iOS test suites PASSED
* 7+1 tvOS test suites PASSED
* 2 demo apps PASSED
* Check new tests
```
    - (void)testGetVariationForFeatureWithFallBackRuleBucketing
    - (void)testGetVariationForFeatureWithFallBackRuleBucketingButNoTargetingRule
    - (void)testGetVariationForFeatureWithFallBackRuleBucketingId
```
PASSED
* Code coverage for
OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m == 92.57%
getVariationForFeatureRollout: == 100%
PASSED

**Comment by kellyroach-optimizely** [comment]
CODE INSPECTION

Revised OPTLYDecisionService.m's getVariationForFeatureRollout:userId:attributes: code
compared with Java reference SDK's getVariationForFeatureInRollout .  The signatures
are aligned and the code is similar as we might expect to see.  APPROVING

**Review by kellyroach-optimizely — APPROVED** [review]
APPROVED

**Pull request merged**